File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3SiriReferenceDataMappingsResponse.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3SiriReferenceDataMappingsResponse {
    /// <summary>
    /// Gets or Sets MappingVersion
    /// </summary>
    [DataMember(Name="mapping_version", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "mapping_version")]
    public string MappingVersion { get; set; }

    /// <summary>
    /// SIRI LineRef
    /// </summary>
    /// <value>SIRI LineRef</value>
    [DataMember(Name="line_refs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "line_refs")]
    public Dictionary<string, V3SiriDirectionRefsDictionary> LineRefs { get; set; }

    /// <summary>
    /// Gets or Sets StopPointRefs
    /// </summary>
    [DataMember(Name="stop_point_refs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "stop_point_refs")]
    public Dictionary<string, V3StopPoint> StopPointRefs { get; set; }

    /// <summary>
    /// Gets or Sets Status
    /// </summary>
    [DataMember(Name="status", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "status")]
    public V3Status Status { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3SiriReferenceDataMappingsResponse {\n");
      sb.Append("  MappingVersion: ").Append(MappingVersion).Append("\n");
      sb.Append("  LineRefs: ").Append(LineRefs).Append("\n");
      sb.Append("  StopPointRefs: ").Append(StopPointRefs).Append("\n");
      sb.Append("  Status: ").Append(Status).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: micronaut/src/main/java/io/swagger/model/V3SiriReferenceDataMappingsResponse.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.model.V3SiriDirectionRefsDictionary;
import io.swagger.model.V3Status;
import io.swagger.model.V3StopPoint;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * V3SiriReferenceDataMappingsResponse
 */
@Validated
@Introspected
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.MicronautCodegen", date = "2024-04-14T03:36:01.894111500+10:00[Australia/Sydney]")
public class V3SiriReferenceDataMappingsResponse   {
  @JsonProperty("mapping_version")
  private String mappingVersion = null;

  @JsonProperty("line_refs")
  @Valid
  private Map<String, V3SiriDirectionRefsDictionary> lineRefs = null;

  @JsonProperty("stop_point_refs")
  @Valid
  private Map<String, V3StopPoint> stopPointRefs = null;

  @JsonProperty("status")
  private V3Status status = null;

  public V3SiriReferenceDataMappingsResponse mappingVersion(String mappingVersion) {
    this.mappingVersion = mappingVersion;
    return this;
  }

  /**
   * Get mappingVersion
   * @return mappingVersion
  **/
  @Schema(readOnly = true, description = "")

  public String getMappingVersion() {
    return mappingVersion;
  }

  public void setMappingVersion(String mappingVersion) {
    this.mappingVersion = mappingVersion;
  }

  public V3SiriReferenceDataMappingsResponse lineRefs(Map<String, V3SiriDirectionRefsDictionary> lineRefs) {
    this.lineRefs = lineRefs;
    return this;
  }

  public V3SiriReferenceDataMappingsResponse putLineRefsItem(String key, V3SiriDirectionRefsDictionary lineRefsItem) {
    if (this.lineRefs == null) {
      this.lineRefs = new HashMap<String, V3SiriDirectionRefsDictionary>();
    }
    this.lineRefs.put(key, lineRefsItem);
    return this;
  }

  /**
   * SIRI LineRef
   * @return lineRefs
  **/
  @Schema(readOnly = true, description = "SIRI LineRef")
  @Valid
  public Map<String, V3SiriDirectionRefsDictionary> getLineRefs() {
    return lineRefs;
  }

  public void setLineRefs(Map<String, V3SiriDirectionRefsDictionary> lineRefs) {
    this.lineRefs = lineRefs;
  }

  public V3SiriReferenceDataMappingsResponse stopPointRefs(Map<String, V3StopPoint> stopPointRefs) {
    this.stopPointRefs = stopPointRefs;
    return this;
  }

  public V3SiriReferenceDataMappingsResponse putStopPointRefsItem(String key, V3StopPoint stopPointRefsItem) {
    if (this.stopPointRefs == null) {
      this.stopPointRefs = new HashMap<String, V3StopPoint>();
    }
    this.stopPointRefs.put(key, stopPointRefsItem);
    return this;
  }

  /**
   * Get stopPointRefs
   * @return stopPointRefs
  **/
  @Schema(readOnly = true, description = "")
  @Valid
  public Map<String, V3StopPoint> getStopPointRefs() {
    return stopPointRefs;
  }

  public void setStopPointRefs(Map<String, V3StopPoint> stopPointRefs) {
    this.stopPointRefs = stopPointRefs;
  }

  public V3SiriReferenceDataMappingsResponse status(V3Status status) {
    this.status = status;
    return this;
  }

  /**
   * Get status
   * @return status
  **/
  @Schema(description = "")

  @Valid
  public V3Status getStatus() {
    return status;
  }

  public void setStatus(V3Status status) {
    this.status = status;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3SiriReferenceDataMappingsResponse v3SiriReferenceDataMappingsResponse = (V3SiriReferenceDataMappingsResponse) o;
    return Objects.equals(this.mappingVersion, v3SiriReferenceDataMappingsResponse.mappingVersion) &&
        Objects.equals(this.lineRefs, v3SiriReferenceDataMappingsResponse.lineRefs) &&
        Objects.equals(this.stopPointRefs, v3SiriReferenceDataMappingsResponse.stopPointRefs) &&
        Objects.equals(this.status, v3SiriReferenceDataMappingsResponse.status);
  }

  @Override
  public int hashCode() {
    return Objects.hash(mappingVersion, lineRefs, stopPointRefs, status);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3SiriReferenceDataMappingsResponse {\n");
    
    sb.append("    mappingVersion: ").append(toIndentedString(mappingVersion)).append("\n");
    sb.append("    lineRefs: ").append(toIndentedString(lineRefs)).append("\n");
    sb.append("    stopPointRefs: ").append(toIndentedString(stopPointRefs)).append("\n");
    sb.append("    status: ").append(toIndentedString(status)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
